heroic-component/src/main/java/com/spotify/heroic/metric/QueryResult.java

```diff
@@ -95,8 +114,51 @@
                 limits.add(ResultLimit.GROUP);
             }
 
-            return new QueryResult(range, groupLimit.limitList(groups), errors, trace,
-                new ResultLimits(limits.build()));
+            QueryResult queryResult = new QueryResult(range, groupLimit.limitList(groups), errors,
+                                                      trace, new ResultLimits(limits.build()));
+
+            queryResult.logQueryInfo(request, requestMetadata,
+                                     logQueries, logQueriesThresholdDataPoints);
+
+            return queryResult;
         };
     }
+
+    public void logQueryInfo(final FullQuery.Request request,
+                             final QueryRequestMetadata requestMetadata,
+                             final boolean logQueries,
+                             final OptionalLimit logQueriesThresholdDataPoints
+    ) {
+        if (!logQueries) {
+            return;
+        }
+
+        totalQueriesProcessed.increment();
+
+        long totDataPoints = 0;
+        for (ShardedResultGroup g : groups) {
+            totDataPoints += g.getMetrics().getData().size();
```

Comment:
I'll clarify that, so that it says "postAggregation..." instead of totDataPoints. 

Also, we were experimenting with taking AggregationResult -> getResults -> getCounters -> get("Aggregation.sampleSize"), and logging that value as preAggregationSampleSize (see the changes in LocalMetricManager and QueryLogger). Would you say that's correct, or is there a better way of getting to know how much data we had before aggregation?